hazelcast/aggregator.py

```diff
@@ -89,245 +88,373 @@ def get_class_id(self):
+
+    Accepts any input value type that is comparable.
 
     Args:
-        attribute_path (str): extracts values from this path if given
+        attribute_path (str): Extracts values from this path, if given.
 
     Returns:
-        Aggregator[int]: an aggregator that calculates the average of the
-        input values
+        Aggregator[any]: An aggregator that calculates the max of the input
+        values.
     """
-    return _LongAverageAggregator(attribute_path)
+    return _MaxAggregator(attribute_path)
 
 
-def long_sum(attribute_path=None):
-    """Creates long sum aggregator.
+def min_(attribute_path=None):
+    """Creates an aggregator that calculates the min of the input values.
 
-    Does NOT accept ``None`` input values.
-    Accepts only long input values (primitive and boxed).
+    Accepts ``None`` input values and ``None`` extracted values.
+
+    Accepts any input value type that is comparable.
 
     Args:
-        attribute_path (str): extracts values from this path if given
+        attribute_path (str): Extracts values from this path, if given.
 
     Returns:
-        Aggregator[int]: an aggregator that calculates the sum of the
-        input values
+        Aggregator[any]: An aggregator that calculates the min of the
+        input values.
     """
-    return _LongSumAggregator(attribute_path)
+    return _MinAggregator(attribute_path)
+
+
+def number_avg(attribute_path=None):
+    """Creates an aggregator that calculates the average of the input values.
+
+    Does NOT accept ``None`` input values or ``None`` extracted values.
+
+    Accepts generic number input values. That means, one should be able to
+    use this aggregator with ``float`` or ``int`` value sent from the Python
+    client unless they are out of range for ``double`` type in Java.
+
+    Args:
+        attribute_path (str): Extracts values from this path, if given.
+
+    Returns:
+        Aggregator[float]: An aggregator that calculates the average of
+        the input values.
+    """
+    return _NumberAverageAggregator(attribute_path)
+
+
+def max_by(attribute_path=None):
+    """Creates an aggregator that calculates the max of the input values
+    extracted from the given ``attribute_path`` and returns the input
+    item containing the maximum value. If multiple items contain the
+    maximum value, any of them is returned.
+
+    Accepts ``None`` input values and ``None`` extracted values.
+
+    Accepts any input value type that is comparable.
+
+    Args:
+        attribute_path (str): Path to extract values from.
+
+    Returns:
+        Aggregator[hazelcast.core.MapEntry]: An aggregator that calculates
+        the input value containing the maximum value extracted from the path.
+    """
+    return _MaxByAggregator(attribute_path)
+
+
+def min_by(attribute_path):
+    """Creates an aggregator that calculates the min of the input values
+    extracted from the given ``attribute_path`` and returns the input
```

Comment:
nit: None of these methods above contained `attribute_path` in the first paragraph of the docstring. Should we remove it from min_by and max_by too?

like this:

```
Creates an aggregator that calculates the min of the input values and ret...``
```

Reply:
No, for them attribute path is optional, but for min_by and max_by it is required. So, I choose to document attribute_path behavior on the parameter docstring for others